### PR TITLE
Remove PySide as a test dependency

### PR DESCRIPTION
**Pull request by mdickinson**
Since #1788 and #1792, we should no longer need a working Qt backend when testing. Accordingly, this PR:

- removes PySide6 as a test dependency
- removes the need to `apt-get` Qt packages on Ubuntu runners
- removes the use of `xvfb-run` on Ubuntu runners

### CODE REVIEW

**Comment by mdickinson** [comment]
This was a bit premature: tests pass, but some of those tests are being skipped. See #1792 for a fix.

**Comment by mdickinson** [comment]
> This was a bit premature: tests pass, but some of those tests are being skipped. See #1792 for a fix.

#1792 was merged; we're back to having no skipped tests.

File: .github/workflows/run-traits-tests.yml [hunk 1]
```diff
@@ -20,19 +15,6 @@ jobs:
...
-        sudo apt-get install libegl1
-        sudo apt-get install libxkbcommon-x11-0
-        sudo apt-get install libxcb-cursor0
-        sudo apt-get install libxcb-icccm4
-        sudo apt-get install libxcb-image0
-        sudo apt-get install libxcb-keysyms1
-        sudo apt-get install libxcb-randr0
-        sudo apt-get install libxcb-render-util0
-        sudo apt-get install libxcb-shape0
-      if: matrix.os == 'ubuntu-latest'
```
Comment (flongford) [inline .github/workflows/run-traits-tests.yml:35]: Very nice tidy up! One less place to remember all the X11 packages...

File: .github/workflows/run-traits-tests.yml [hunk 2]
```diff
@@ -44,9 +26,6 @@ jobs:
...
         mkdir testdir
-    - name: Test Traits package (Linux)
-      run: cd testdir && xvfb-run -a python -m unittest discover -v traits
-      if: matrix.os == 'ubuntu-latest'
-    - name: Test Traits package (Windows/macOS)
-      run: cd testdir && python -m unittest discover -v traits
-      if: matrix.os != 'ubuntu-latest'
+    - name: Test Traits package
+      run: python -m unittest discover -v traits
+      working-directory: testdir
```
Comment (flongford) [inline .github/workflows/run-traits-tests.yml:31]: Huh, TIL [`working-directory`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#defaultsrunworking-directory) context in GH Actions.

Quite handy indeed.

**Review by flongford — APPROVED** [review]
LGTM!

**Comment by mdickinson** [comment]
Thanks for review, @flongford!

**Pull request merged**